view/next-project/src/pages/purchase_report_list/index.tsx

```diff
@@ -56,17 +63,31 @@ export default function PurchaseReports() {
   }, [buyReports]);
 
   const updateSealCheck = (id: number) => {
-    setSealChecks((prev) => ({
-      ...prev,
-      [id]: !prev[id],
-    }));
+    setSealChecks((prev) => {
+      const newValue = !prev[id];
+      if (!newValue) {
+        setSettlementChecks((prevSettlement) => ({
+          ...prevSettlement,
+          [id]: false,
+        }));
+      }
+      return {
+        ...prev,
+        [id]: newValue,
+      };
+    });
```

Comment:
setの中でまた別のsetを行うとちょっと読みにくいかも、、
外に出せようやし、二重否定ならそのままsealChecks[id]使っちゃって良さそう
```suggestion
    if (sealChecks[id]) {
      setSettlementChecks((prevSettlement) => ({
        ...prevSettlement,
        [id]: false,
      }));
    }

    setSealChecks((prev) => {
      return {
        ...prev,
        [id]: !prev[id],
      };
    });
```